File: Kattis/C++/Eating-out/Eating-out.cpp
#include <bits/stdc++.h>

using namespace std;
using i64 = long long;

int main()
{
    i64 a,b,c,m;
    cin >> m >> a >> b >> c;

    if((a + b + c) / 2 <= m)
    {
        cout << "possible" << endl;
    }
    else
    {
        cout << "impossible" << endl;
    }
}

File: Kattis/Python/Eating-out.py

def main():
    m,a,b,c = input().split()

    m = int(m)
    a = int(a)
    b = int(b)
    c = int(c)

    if (a + b + c)/2 <= m:
        print("possible\n")
    else:
        print("impossible\n")

if __name__ == "__main__":
    main()
